includes/spoolchanges.js

```diff
@@ -62,34 +63,55 @@ module.exports = function(db, log, bufferSize, ee, callback) {
   };
 
   // stream the changes feed to disk
-  var changesRequest = db.changesAsStream({ seq_interval: 10000 })
-    .on('error', function(err) {
-      callback(new error.BackupError('SpoolChangesError', `Failed changes request - ${err.message}`));
-    })
-    .on('response', function(resp) {
-      if (resp.statusCode >= 400) {
-        changesRequest.abort();
-        callback(error.convertResponseError(resp));
+  db.service.postChangesAsStream({ db: db.db, seq_interval: 10000 }).then(response => {
+    response.result.pipe(liner())
+      .on('error', function(err) {
+        callback(err);
+      })
+      .pipe(change(onChange))
+      .on('error', function(err) {
+        callback(err);
+      })
+      .on('finish', function() {
+        processBuffer(true);
+        if (!lastSeq) {
+          logStream.end();
+          debug('changes request terminated before last_seq was sent');
+          callback(new error.BackupError('SpoolChangesError', 'Changes request terminated before last_seq was sent'));
+        } else {
+          debug('finished streaming database changes');
+          logStream.end(':changes_complete ' + lastSeq + '\n', 'utf8', callback);
+        }
+      });
+  }).catch(err => {
+    if (err.code && err.code >= 400) {
+      if (err.body) {
+        class ResponseWriteable extends stream.Writable {
```

Comment:
A comment should be good here, what this is for. For _bulk_docs there was one previously:
```
    // Class for streaming _bulk_docs responses into
    // In general the response is [] or a small error/reason JSON object
    // so it is OK to have this in memory.
```

Reply:
Done in 290848be044e9fffd74297fdb1234c10b12b723c - I also pulled the class out of the function for a readability improvement.